include/transport/time_queue.h

```diff
@@ -232,16 +233,22 @@ namespace qtransport {
          * @param interval                  Interval size of each bucket. Value must be > 0 and != duration
          * @param timer                     Shared pointer to timer service
          * @param initial_queue_size        Initial size of the queue to reserve.
+         * @param spike_period_interval     Spike period interval defines the time allowed for one spike, ZERO disables spikes
+         * @param spike_duration            Spike allowed duration, max time to allow spike in ticks
          *
          * @throws std::invalid_argument    If the duration or interval do not meet requirements.
          * @throws std::runtime_error       If the timer is null.
          */
         time_queue(size_t duration,
                    size_t interval,
                    std::shared_ptr<queue_timer_service> timer,
-                   size_t initial_queue_size)
-          : time_queue(duration, interval, timer)
+                   size_t initial_queue_size,
+                   size_t spike_period_interval=30000,
```

Comment:
Could we update the comment to specify the units of time here .. Is it 30 milliseconds ?

Reply:
+1

Maybe even include it in the name of the param: `spike_period_interval_ns` (or whatever the unit is)

Reply:
It's relative to the time_queue constructor params `interval` and `duration`.  The value of these settings, including TTL is actually the count of interval.  The default is 1 and the default duration is ms, so these values, including TTL is (value * 1ms).  In this case, changing the param to ms would be fine, but then it would only be fine as long as you never changed those.

Priority queue, for example, locks in 1ms, so those settings are in ms.   Time-queue is customizable and therefore the values are relative to the constructor params. For now, I'll add some comments around this.